src/utilities.jl

```diff
@@ -3,6 +3,42 @@
 # Utilities.
 #
 
+#
+# Expression Capture.
+#
+
+"""
+    interpolation(object::T, captured::Expr) -> new_object
+
+Interface method for hooking into interpolation within docstrings to change
+the behaviour of the interpolation. `object` is the interpolated object within a
+docstring and `captured` is the raw expression that is documented by the docstring
+in which the interpolated `object` has been included.
+
+To define custom behaviour for your own `object` types implement a method of
+`interpolation(::T, captured)` for type `T` and return a `new_object` to
+be interpolated into the final docstring. Note that you must own the definition
+of type `T`. `new_object` does not need to be of type `T`.
+"""
+interpolation(@nospecialize(object), @nospecialize(_)) = object
+
+# During macro expansion process the interpolated string and replace all interpolation
+# syntax with calls to `interpolation` that pass through the documented expression along
+# with the resolved object that was interpolated.
+function _capture_expression(docstr::Expr, expr::Expr)
+    if Meta.isexpr(docstr, :string)
```

Comment:
Out of curiosity, when is this not the case?

Reply:
Since you can do ridiculous things like

```
julia> @doc 1 f(x) = x
f

help?> f
search: f fd for fma fld fld1 fill fdio frexp foldr foldl flush floor float first fill! fetch fldmod filter falses finally

1-element Vector{Int64}:
 1
```

and just attach random objects as the docstring, none of which can be supported by the expression capture code here. Only `Expr(:string)` can. If I recall correctly that was supported since PyCall.jl wanted to support retrieving docstrings from Python or something.